useful-scripts/misc-test-utils/src/services/commonService.ts

```diff
@@ -0,0 +1,202 @@
+/* eslint-disable no-console */
+import axios from 'axios';
+import chalk from 'chalk';
+import StreamZip from 'node-stream-zip';
+import globby from 'globby';
+import fs from 'fs';
+import rimraf from 'rimraf';
+import { projectRoot } from '../config';
+import errorHandler from '../utils/errorHandler';
+import { ReleaseData } from '../types/ReleaseData';
+import sleep from '../utils/sleep';
+
+const { ADMIN_URL, JWT_TOKEN } = process.env;
+
+const cwd = projectRoot;
+
+const commonService = {
+  validate: async () => {
+    const zipGlob = await globby(`${cwd}/*.zip`);
+    if (fs.existsSync(zipGlob[0]) && zipGlob[0].includes('archive.zip')) {
+      console.log(chalk.green('found zip file, continuing'));
+    } else {
+      throw new Error(
+        'No zip file named archive.zip in root dir! Exiting test with failures',
+      );
+    }
+
+    if (fs.existsSync(`${cwd}/test-files`)) {
+      rimraf(`${cwd}/test-files/*`, () => {
+        console.log(chalk.green('cleaned test-files folder'));
+      });
+    }
+
+    if (fs.existsSync(`${cwd}/zip-files`)) {
+      rimraf(`${cwd}/zip-files/*`, () => {
+        console.log(chalk.green('cleaned zip-files folder'));
+      });
+    }
+
+    if (!fs.existsSync(`${cwd}/test-files`)) {
+      fs.mkdirSync(`${cwd}/test-files`);
+    }
+
+    if (!fs.existsSync(`${cwd}/zip-files`)) {
+      fs.mkdirSync(`${cwd}/zip-files`);
+    }
+
+    if (zipGlob[1]) {
+      rimraf(zipGlob[1], () => {
+        console.log(chalk.green('cleaned stale zip files'));
+      });
+    }
+  },
+  createRelease: async (publicationId: string): Promise<string | null> => {
+    console.time('createRelease');
+    try {
+      const res = await axios({
+        method: 'POST',
+        url: `${ADMIN_URL}/api/publications/${publicationId}/releases`,
+        data: {
+          timePeriodCoverage: { value: 'AY' },
+          releaseName: 2222,
+          typeId: '1821abb8-68b0-431b-9770-0bea65d02ff0',
+          publicationId: 'c76c1bc4-0a01-4116-74cd-08d8eebbf0c1',
+          templateReleaseId: '',
+        },
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${JWT_TOKEN}`,
+        },
+      });
+
+      console.timeEnd('createRelease');
+
+      const releaseId = res.data.id;
+
+      console.log(
+        chalk.green(
+          `Release URL: ${ADMIN_URL}/publication/${publicationId}/release/${releaseId}/data`,
+        ),
+      );
+      return releaseId;
+    } catch (e) {
+      errorHandler(e);
+    }
+    return null;
+  },
+
+  extractZip: async () => {
```

Comment:
Could make this method more generic so that it could handle other cases?

Reply:
Going to leave this one for now. It'll be a good option to have but it'll require a fair bit of refactoring to handle this. I'll make a note of it so I can pick it up in the future